collective-big-count/test_allgatherv.c

```diff
@@ -0,0 +1,388 @@
+    // Adjust these to be V_SIZE_INT - displacement strides so it will pass
+    ret += my_c_test_core(MPI_INT,
+                          (proposed_count - disp_stride*world_size) * (size_t)world_size,
+                          MODE_SKIP, true, true);
+
+    // Note: Displacement is an int, so the recv buffer cannot be too large as to overflow the int
+    // As such divide by the world_size
+    proposed_count = calc_uniform_count(sizeof(double _Complex), TEST_UNIFORM_COUNT / (size_t)world_size,
+                                        (size_t)world_size, (size_t)world_size, 1.0);
+    ret += my_c_test_core(MPI_C_DOUBLE_COMPLEX, proposed_count * (size_t)world_size, MODE_PACKED, true,
+                          true);
+    // Adjust these to be V_SIZE_INT - displacement strides so it will pass
+    ret += my_c_test_core(MPI_C_DOUBLE_COMPLEX,
+                          (proposed_count - disp_stride*world_size) * (size_t)world_size,
+                          MODE_SKIP, true, true);
+    if (allow_nonblocked) {
+        proposed_count = calc_uniform_count(sizeof(int), TEST_UNIFORM_COUNT / (size_t)world_size,
+                                            (size_t)world_size, (size_t)world_size, 1.0);
+        ret += my_c_test_core(MPI_INT, proposed_count * (size_t)world_size, MODE_PACKED, true, 
+                              false);
+        // Adjust these to be V_SIZE_INT - displacement strides so it will pass
+        ret += my_c_test_core(MPI_INT,
+                              (proposed_count - disp_stride*world_size) * (size_t)world_size,
+                              MODE_SKIP, true, false);
+        proposed_count = calc_uniform_count(sizeof(double _Complex), TEST_UNIFORM_COUNT / (size_t)world_size,
+                                            (size_t)world_size, (size_t)world_size, 1.0);
+        ret += my_c_test_core(MPI_C_DOUBLE_COMPLEX, proposed_count * (size_t)world_size, MODE_PACKED,
+                              true, false);
+        // Adjust these to be V_SIZE_INT - displacement strides so it will pass
+        ret += my_c_test_core(MPI_C_DOUBLE_COMPLEX,
+                              (proposed_count - disp_stride*world_size) * (size_t)world_size,
+                              MODE_SKIP, true, false);
+    }
+#endif
+
+    /*
+     * All done
+     */
+    MPI_Finalize();
+    return ret;
+}
+
+int my_c_test_core(MPI_Datatype dtype, size_t total_num_elements, int mode, bool in_place, bool blocking)
+{
+    int ret = 0;
+    size_t i;
+
+    // Actual payload size as divisible by the sizeof(dt)
+    size_t payload_size_actual;
+
+    /*
+     * Initialize vector
+     */
+    int *my_int_send_vector = NULL;
+    int *my_int_recv_vector = NULL;
+    int int_exp;
+
+    double _Complex *my_dc_send_vector = NULL;
+    double _Complex *my_dc_recv_vector = NULL;
+    double _Complex dc_exp;
+
+    int *my_recv_counts = NULL;
+    int *my_recv_disp = NULL;
+    int send_count = 0;
```

Comment:
We want to keep `send_count` an `int` here since we pass it directly into the `MPI_Allgatherv`. We have an assert check before we set it to ensure it is within the `INT_MAX` range.